checks.d/spark.py

```diff
@@ -228,32 +232,48 @@ def _get_running_apps(self, instance, tags):
         '''
         Determine what mode was specified
         '''
+
+        # Get the master address from the instance configuration
+        master_address = instance.get(MASTER_ADDRESS)
+        if master_address is None:
+            master_address = instance.get(DEPRECATED_MASTER_ADDRESS)
+
+            if master_address:
+                self.log.warning('The use of `%s` is deprecated. Please use `%s` instead.' %
+                    (DEPRECATED_MASTER_ADDRESS, MASTER_ADDRESS))
+            else:
+                raise Exception('A URL for `%s` must be specified in the instance '
+                    'configuration' % MASTER_ADDRESS)
+
+        # Get the cluster name from the instance configuration
+        cluster_name = instance.get('cluster_name')
```

Comment:
The reason I decided to require it is because Spark Standalone is still running on a cluster it is just a Spark only cluster. It is also a good way for anyone who may be running multiple instances that are executing a job with the same name to differentiate metrics. Also with my next PR (#2797) it is required, so I felt it was better to make it required for any mode of spark you chose: Standalone, YARN, or Mesos.